src/scripts/release.zig

```diff
@@ -382,9 +380,102 @@ fn build_tigerbeetle(shell: *Shell, info: VersionInfo, dist_dir: std.fs.Dir) !vo
                 shell.project_root,
                 "tigerbeetle-" ++ target,
             );
+            try shell.project_root.deleteFile("tigerbeetle");
+        }
+
+        const past_versions_aarch64 = try build_multiversion_body(
+            shell,
+            "aarch64-macos",
+            debug,
+            "multiversion-build/multiversion-aarch64-macos" ++ debug_suffix ++ ".body",
+        );
+
+        const past_versions_x86_64 = try build_multiversion_body(
+            shell,
+            "x86_64-macos",
+            debug,
+            "multiversion-build/multiversion-x86_64-macos" ++ debug_suffix ++ ".body",
+        );
+
+        try macos_universal_binary_build(
+            shell,
+            "multiversion-build/tigerbeetle-macos-empty-headers" ++ debug_suffix,
+            &.{
+                .{
+                    .cpu_type = std.macho.CPU_TYPE_ARM64,
+                    .cpu_subtype = std.macho.CPU_SUBTYPE_ARM_ALL,
+                    .path = "tigerbeetle-aarch64-macos",
+                },
+                .{
+                    .cpu_type = std.macho.CPU_TYPE_X86_64,
+                    .cpu_subtype = std.macho.CPU_SUBTYPE_X86_64_ALL,
+                    .path = "tigerbeetle-x86_64-macos",
+                },
+                .{
+                    .cpu_type = 0x00000001, // VAX == .tb_mvb for aarch64
+                    .cpu_subtype = 0x00000000,
+                    .path = "multiversion-build/multiversion-aarch64-macos" ++ debug_suffix ++
+                        ".body",
+                },
+                .{
+                    .cpu_type = 0x00000002, // ROMP == .tb_mvh for aarch64
+                    .cpu_subtype = 0x00000000,
+                    .path = "multiversion-build/multiversion-empty.header",
+                },
+                .{
+                    .cpu_type = 0x00000004, // NS32032 == .tb_mvb for x86_64
+                    .cpu_subtype = 0x00000000,
+                    .path = "multiversion-build/multiversion-x86_64-macos" ++ debug_suffix ++
+                        ".body",
+                },
+                .{
+                    .cpu_type = 0x00000005, // NS32332 == .tb_mvh for x86_64
+                    .cpu_subtype = 0x00000000,
```

Comment:
What is the meaning of these "cpu types"? Does MachO just not care what the values are, so they are arbitrary constants from tigerbeetle's point-of-view?

Could we define them as an `enum(c_int) { aarch64_tb_mvh = 1, aarch64_tb_mvb = 2, ... }`, since they are used in multiple places?

Also, maybe in `parse_macho` we could use a `std.enums.EnumSet` to verify that we find (exactly one of) each section in the header.

Reply:
> What is the meaning of these "cpu types"? Does MachO just not care what the values are, so they are arbitrary constants from tigerbeetle's point-of-view?

Exactly that, they are valid (in the spec) but ancient (macOS has never run on anything other than x86_64 / arm64) platforms. They were chosen so that it wouldn't be a random value, but also wouldn't be something that we could realistically encounter.

Reply:
> Also, maybe in parse_macho we could use a std.enums.EnumSet to verify that we find (exactly one of) each section in the header.

I think we might go the opposite way in future: allow non-universal binaries, so not too sure we want to enforce this.